src/main/java/se/iths/jakartaeelab1/exceptions/ExceptionMapper.java

```diff
@@ -0,0 +1,43 @@
+package se.iths.jakartaeelab1.exceptions;
+
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.Provider;
+
+@Provider
+public class ExceptionMapper implements jakarta.ws.rs.ext.ExceptionMapper<Exception> {
+
+    @Override
+    public Response toResponse(Exception exception) {
+        ErrorResponse errorResponse = new ErrorResponse();
+        errorResponse.setError(exception.getMessage());
+        errorResponse.setCode(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+
+        return Response
+                .status(Response.Status.INTERNAL_SERVER_ERROR)
+                .entity(errorResponse)
+                .type(MediaType.APPLICATION_JSON)
+                .build();
+    }
+
+    public static class ErrorResponse {
+        private int code;
+        private String error;
+
+        public int getCode() {
+            return code;
+        }
+
+        public void setCode(int code) {
+            this.code = code;
+        }
+
+        public String getError() {
+            return error;
+        }
+
+        public void setError(String error) {
+            this.error = error;
+        }
+    }
+}
```

Comment:
After looking at Martins project, it is a better way to solve the issue of formatting the exceptions and what the message the user will recieve.
```suggestion
package se.iths.jakartaeelab1.exceptions;

import jakarta.json.Json;
import jakarta.json.JsonObject;
import jakarta.validation.ConstraintViolation;
import jakarta.validation.ConstraintViolationException;
import jakarta.ws.rs.core.Context;
import jakarta.ws.rs.core.Response;
import jakarta.ws.rs.core.UriInfo;
import jakarta.ws.rs.ext.ExceptionMapper;
import jakarta.ws.rs.ext.Provider;

import java.util.Set;

@Provider
public class ConstraintViolationExceptionMapper implements ExceptionMapper<ConstraintViolationException> {
    @Context
    UriInfo uriInfo;
    @Override
    public Response toResponse(ConstraintViolationException exception) {
        Set<ConstraintViolation<?>> constraintViolations = exception.getConstraintViolations();

        final var jsonObjectBuilder = Json
                .createObjectBuilder()
                .add("host",uriInfo.getAbsolutePath().getHost())
                .add("resource",uriInfo.getAbsolutePath().getPath())
                .add("title","Validation Errors");

        final var jsonArray = Json.createArrayBuilder();

        for (final var constraint : constraintViolations) {
            String message = constraint.getMessage();
            String violatedField = constraint.getPropertyPath().toString().split("\\.")[2];

            JsonObject jsonError = Json.createObjectBuilder()
                    .add("field",violatedField)
                    .add("violationMessage",message)
                    .build();
            jsonArray.add(jsonError);
        }
        JsonObject errorJsonEntity = jsonObjectBuilder.add("errors",jsonArray.build()).build();
        return Response.status(Response.Status.BAD_REQUEST).entity(errorJsonEntity).build();
    }
}

```

Reply:
Okay, I'll check it out and update 👍

Reply:
Does it not work even with the path in HelloResource? (test-exception)

Reply:
Yes it does, I got it to work now. Don't know how it didn't before. But it gives us code 500 every time. Which is not what we want.
![image](https://github.com/Win-ther/JakartaEELab1/assets/143023413/9eb06525-4587-40c0-8d31-8da8ea3195a9)

It is better if we get 404 Not Found when we enter incorrect URI. This is what we get without your ExceptionMapper.
![image](https://github.com/Win-ther/JakartaEELab1/assets/143023413/c7b9f779-4b37-4046-8bcf-4d2e29082c6a)